### ISSUE TITLE
Fatal Crash on Intel iGPU, inconsistently crashes entire OS

### ISSUE THREAD

**Issue by BobGoldham** [hyprwm/Hyprland#6346]
### System Info and Version

<details> <summary>System/Version info</summary>
hyprctl unusable as Hyprland won't start in the first place. Please provide further instructions on what to put here.
</details>

### Bug or Regression?

Bug

### Description

Attempting to start Hyprland either yields
crash1 or, when manually linking my GPU to `.config/hypr/card`, crash2.

I have explicitly set the kernel parameter `i915.modeset=1`

Most of the time, though not always (and I can't seem to identify what determines this), it also crashes my entire laptop into a state where the monitor is off and capslock key blinking. The laptop becomes trapped in that state for a while until eventually the BIOS resets itself, which leads to me having to re-adjust the clock time and boot settings. I have attached a core dump that I *think* is related to these occurences, but I'm not sure I've identified that correctly.

sway crashes with similar issue, weston works fine. I can start Hyprland from within weston. This leads me to believe that the issue is related to how WLRoots handles DRM devices.

### How to reproduce

- Attempt to start Hyprland?

```
$ lspci | grep 'VGA'
00:02.0 VGA compatible controller: Intel Corporation Iris Plus Graphics G1 (Ice Lake) (rev 07)
```

### Crash reports, logs, images, videos

[crash1](https://github.com/user-attachments/files/15614211/hyprlandCrashReport14157.txt)
[crash2](https://github.com/user-attachments/files/15614223/hyprlandCrashReport919.txt)
[coredump.txt](https://github.com/user-attachments/files/15614282/coredump.txt)

**Comment by MightyPlaza** [comment]
`[backend/session/session.c:407] Unable to open /home/user/.config/hypr/card as DRM device`
that's on both crashes and I don't think it's right

**Comment by BobGoldham** [comment]
> `[backend/session/session.c:407] Unable to open /home/user/.config/hypr/card as DRM device` that's on both crashes and I don't think it's right

I agree that doesn't seem right. I have attempted both not providing a symlink, providing a symlink to `/dev/dri/by-path/...-card` and (for completeness' sake and because I figured it wouldn't do anything *worse* at least) `/dev/dri/by-path/...-render`. 

None of these seem to be recognized as a valid DRM device.

**Comment by MightyPlaza** [comment]
why do you keep providing the drm device it should pick it automatically
please send the crash report without setting `WLR_DRM_DEVICES`

**Comment by BobGoldham** [comment]
The envvar `WLR_DRM_DEVICES` is unset.
Not linking anything to `.config/hypr/card` crashes with the same symptoms (screen goes black, capslock blinking, bios resets), but yields no logs in `~/.cache/hypr`. It appears that the OS crashes "too fast" for the log to write?

I also find it highly concerning that this doesn't just crash the compositor, but my entire OS and apparently, somehow, resets my UEFI settings!?

`journalctl -b 1 -x` has the following to say about that 
[coredump.txt](https://github.com/user-attachments/files/15614654/coredump.txt)

**Comment by MightyPlaza** [comment]
can you try starting from a tty without lightdm?

**Comment by BobGoldham** [comment]
That's what I have been doing. My DM was simply running in another tty. Disabling and stopping it then attempting to start Hyprland yields the same results.

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
